tools/storage-advisor/src/s3Issues.go

```diff
@@ -373,68 +531,181 @@ func runS3Checks(cfg aws.Config, err error) {
 			buckets = append(buckets, Bucket{Name: bucket})
 		}
 	}
-	//buckets = buckets[0:10]
-
-	// Print the bucket names.
-	//	log.Println("S3 Prefixies:")
-	//	for _, name := range bucketNames {
-	//		log.Println(name)
-	//	}
-
-	// Create an S3 service client.
+	buckets = buckets[0:10]
 
 	for i := range buckets {
 		// Get bucket region
 		r, err := client.GetBucketLocation(context.TODO(), &s3.GetBucketLocationInput{
 			Bucket: &buckets[i].Name,
 		})
 		if err != nil {
-			buckets[i].Region = err.Error()
+			buckets[i].Region = "Error" //err.Error()
 		} else {
 			buckets[i].Region = string(r.LocationConstraint)
 		}
 	}
 
-	checkAclSettings(buckets, client)
+	issues := make([]FindingWithBucket, 0)
 
-	checkVersioning(buckets, client)
+	checkAclSettings(buckets, client, &issues)
 
-	checkEncryption(buckets, client)
+	checkVersioning(buckets, client, &issues)
 
-	/*
+	checkEncryption(buckets, client, &issues)
 
-		checkInventory(buckets, client)
+	checkLogging(buckets, client, &issues)
 
-		for i := range buckets {
-			// Get value of bucket's server access logging
-			r, err := client.GetBucketLogging(context.TODO(), &s3.GetBucketLoggingInput{
-				Bucket: &buckets[i].Name,
-			})
-			if err != nil {
-				buckets[i].Logging = err.Error()
-			} else {
-				if r.LoggingEnabled == nil {
-					buckets[i].Logging = "disabled"
-				} else {
-					buckets[i].Logging = fmt.Sprintf("%s/%s", *r.LoggingEnabled.TargetBucket, *r.LoggingEnabled.TargetPrefix)
-				}
-			}
+	checkInventory(buckets, client, &issues)
+
+	writeBasicIssues(issues, identity)
+
+	if api.IsConfigured() {
```

Comment:
Is it expected by the caller that nothing happens if a token is undefined?

Reply:
Yes, it is expected behaviour for now.